File: Java/Lists/PlayerMusic/Player.java
package Lists.PlayerMusic;

public class Player {
    Node<String> musicNode;
    DoubleLinkedList<String> playlist = new DoubleLinkedList<>();

    public void addMusic(String name) {
        Node<String> music = new Node<>(name);
        playlist.append(music);
    }

    public void play() {
        if (playlist.isEmpty()) {
            System.out.println("Playlist vazia!");
            return;
        }

        /*
         * while (musicNode != null) {
         * System.out.println("Tocando agora" + musicNode.data);
         * try {
         * Thread.sleep(2000);
         * } catch (InterruptedException e) {
         * System.out.println("Opa, Playes interrompido");
         * }
         * musicNode = musicNode.next;
         * }
         */

        musicNode = this.playlist.head;
        System.out.println("▶s Play: " + musicNode.data);
    }

    public void next() {
        if (playlist.isEmpty())
            return;

        if (musicNode.next != null) {
            setMusicNode(musicNode.next);
        } else {
            musicNode = playlist.head;
        }
        System.out.println("⏭ Next: " + musicNode.data);

    }

    public void previous() {
        if (playlist.isEmpty())
            return;

        if (musicNode.prev != null) {
            setMusicNode(musicNode.prev);
        } else {
            musicNode = playlist.tail;
        }
        System.out.println("⏮ Previous: " + musicNode.data);

    }

    public Node<String> getMusicNode() {
        return musicNode;
    }

    public void setMusicNode(Node<String> musicNode) {
        this.musicNode = musicNode;
    }

}

File: Python/Lists/Player.py
from DoubleLinkedList import DoubleLinkedList
from Node import Node

class Player:
    def __init__(self):
        self.playlist = DoubleLinkedList[str]()
        self.musicNode = None

    def addMusic(self, name: str):
        music = Node(name)
        self.playlist.append(music)
    
    def play(self):
        if(self.playlist.isEmpty()):
            print("This playlist is empty")
            return
        
        self.musicNode = self.playlist.head
        print(f"▶ Play: {self.musicNode.data}" )

    def next(self):
        if(self.playlist.isEmpty()):
            print("This list is empty")
            return
        
        if(self.musicNode.next != None):
            self.musicNode = self.musicNode.next
        else:
            self.musicNode = self.playlist.head
        
        print(f"Next music is: {self.musicNode.data}")

    def previous(self):
        if(self.playlist.isEmpty()):
            print("This list is empty")
            return

        if(self.musicNode.prev != None):
            self.musicNode = self.musicNode.prev
        else:
            self.musicNode = self.playlist.tail

        print(f"Previous music is: {self.musicNode.data}")
     

    def setMusic(self, musicNode: Node):
        self.musicNode = musicNode
    
    def getMusic(self):
        return self.music.data


